File: java/scripts/HttpClient.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

public class HttpClient {
    public static void main(String[] args) {
        try {
            HttpClient client = HttpClient.newBuilder()
                .connectTimeout(Duration.ofSeconds(10))
                .build();

            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create("https://api.github.com/users/github"))
                .header("User-Agent", "Java HttpClient")
                .GET()
                .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            System.out.println("Status Code: " + response.statusCode());
            System.out.println("Response Body:\n" + response.body());

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
        }
    }
}

File: ruby/scripts/http_client.rb
#!/usr/bin/env ruby

require 'net/http'
require 'json'
require 'uri'

url = URI('https://api.github.com/users/github')

begin
  response = Net::HTTP.start(url.host, url.port, use_ssl: true) do |http|
    request = Net::HTTP::Get.new(url)
    request['User-Agent'] = 'Ruby HTTP Client'
    http.request(request)
  end

  puts "Status Code: #{response.code}"

  if response.code == '200'
    data = JSON.parse(response.body)
    puts "\nUser: #{data['name']}"
    puts "Bio: #{data['bio']}"
    puts "Public repos: #{data['public_repos']}"
  end
rescue StandardError => e
  puts "Error: #{e.message}"
end
